central/sensor/service/pipeline/clustermetrics/update_maxima.go

```diff
@@ -14,15 +14,40 @@ type BillingMetrics struct {
 }
 
 var (
+	// lastKnown stores the last known metrics per cluster.
+	lastKnown = map[string]BillingMetrics{}
+
+	// nodesMap and coresMap store the maximum numbers of nodes and cores
+	// per cluster.
 	nodesMap = maputil.NewMaxMap[string, int64]()
 	coresMap = maputil.NewMaxMap[string, int64]()
 )
 
 func updateMaxima(clusterID string, cm *central.ClusterMetrics) {
+	lastKnown[clusterID] = BillingMetrics{
```

Comment:
`updateMaxima` is called from the sensor pipeline, right? I don't think it's concurrency safe to write to this variable without a mutex.

Reply:
My bad, I forgot to protect the map.